### DISCUSSION TITLE
how does freshRSS handle server outages/ downtime

### DISCUSSION THREAD

**Discussion by OneWeekNotice** [FreshRSS/FreshRSS#6772] [Q&A] [answered]
Hello,

Was wondering how freshRSS handles server outages /downtime. 

Example:  server crashes/ power goes out/ server gets disconnected from the Internet, human error, anything that would bring the server down for lets say a couple of days to a week.

When the server comes back online, will it be able to populate/pull all feeds from the time it went down?

Or will it only pull the latest and any feeds that push/updated while the server was down will be lost/ not pulled?

I guess I'm asking how does it manage feed pulling/updating. For example does it have a marker/last known location and pull everything that it missed? 

If yes, would there ever be a case where it will miss anything?

Thanks in advance for any guidance/ answering any questions

**Comment by Frenzie** [comment]
It will pull anything it couldn't eventually (since there's a back off mechanism) or you can just trigger it manually. But feeds might only offer a few entries,  in which case of course anything missed is missed forever.

**Reply by OneWeekNotice** [reply]
Thanks for the clarification. Out of curiosity, when does the back off mechanism occur/trigger?

**Reply by Frenzie** [reply]
I don't recall the specifics otoh but iirc a 404 reduces the update frequency to at most once an hour.

**Reply by Alkarex** [reply, marked as answer]
I do not think we have anything special for errors, @Frenzie , besides the normal TTL.
When FreshRSS is up again, it will resume retrieving the feeds according to the cron and caching rules.
The amount of articles potentially lost depends on the duration of the outage, and on which Web sites you subscribe to.
Indeed, some Web sites keep many articles in their RSS feeds (which can correspond to days or even weeks worth of news), while some other Web sites publishing articles at a high frequency might be more easily impacted.

**Reply by Frenzie** [reply]
I think it's SimplePie that does it, but it definitely does something. Back off may not have been the right word choice.

**Comment by OneWeekNotice** [comment]
Thanks very much for the answer. To summary

- freshRSS will re-sync with the feeds/articles whenever it comes back online
- Articles being lost is dependent on source RSS feed management.  Example: how long they keep their articles on their RSS feed. (where freshRSS will pull the latest articles that were not pulled)
